docker-nginx/pages/loading.html

```diff
@@ -0,0 +1,256 @@
+        }
+
+        .text-center {
+            text-align: center !important;
+        }
+
+        @media (min-width: 992px) {
+
+            .container,
+            .container-lg,
+            .container-md,
+            .container-sm {
+                max-width: 960px;
+            }
+        }
+
+        @media (min-width: 768px) {
+
+            .container,
+            .container-md,
+            .container-sm {
+                max-width: 720px;
+            }
+        }
+
+        @media (min-width: 576px) {
+
+            .container,
+            .container-sm {
+                max-width: 540px;
+            }
+        }
+
+        .container,
+        .container-fluid,
+        .container-lg,
+        .container-md,
+        .container-sm,
+        .container-xl {
+            width: 100%;
+            padding-right: 15px;
+            padding-left: 15px;
+            margin-right: auto;
+            margin-left: auto;
+        }
+
+        /* added qfc style */
+        a {
+            color: #4a6fae;
+            text-decoration: none;
+        }
+
+        a :hover {
+            color: #3f5e93;
+        }
+
+        .btn-primary {
+            color: #fff;
+            background-color: #4a6fae;
+            border-color: #4a6fae;
+        }
+
+        .btn-primary:hover {
+            color: #fff;
+            background-color: #3f5e93;
+            border-color: #3b588a;
+        }
+
+        /* Add spinner animation */
+        .spinner {
+            width: 40px;
+            height: 40px;
+            margin: 20px auto;
+            border: 4px solid #f3f3f3;
+            border-top: 4px solid #4a6fae;
+            border-radius: 50%;
+            animation: spin 1s linear infinite;
+        }
+
+        @keyframes spin {
+            0% { transform: rotate(0deg); }
+            100% { transform: rotate(360deg); }
+        }
+    </style>
+
+<script>
+    /**
+     * This function implements a retry mechanism for loading pages with a maximum retry limit.
+     * It uses sessionStorage to store the retry count across page reloads.
+     *
+     * The retry logic works as follows:
+     * 1. First attempt happens immediately when loading page is shown
+     * 2. Then retries every 10 seconds, up to 5 more times (60 seconds total)
+     * 3. If still failing after all retries, redirects to 500 error page
+     *
+     * The retry count is stored in sessionStorage to persist across page reloads.
+     */
+    (function() {
+        const refreshCountKey = 'qfcLoadingRefreshCount';
+        const maxRefreshes = 6; // 6 refreshes * 10 seconds per refresh = 60 seconds
+        const errorPageUrl = '/pages/500.html';
+        const retryUrl = '/'; // The URL to navigate to for a retry (e.g., the root)
+        const retryIntervalMs = 10000; // 10 seconds
+
+        const rawCount = sessionStorage.getItem(refreshCountKey);
+        let currentRefreshCount = parseInt(rawCount || '0', 10);
+        currentRefreshCount++;
```

Comment:
```suggestion
        const incrementedRefreshCount = parseInt(rawCount || '0', 10) + 1;
```
Less mutable state, less problems :)

Reply:
@toebivankenoebi you need to change the variable also a few line below where it is actually used.

This is a good note that I don't always propose commitablr code, I just give a sign what can/should be changed and it's up to the developer to pick the important bits and apply across the code.

Reply:
Will do and test the behaviour properly again before pushing.

Reply:
Done and successfully tested.

Reply:
Thanks!